bindings/Sofa/tests/Core/BaseData.py

```diff
@@ -408,3 +408,24 @@ def test_readOnly(self):
         self.assertFalse(data.isReadOnly())
         data.setReadOnly(True)
         self.assertTrue(data.isReadOnly())
+
+    def test_DataContainer_resize(self):
+        root = Sofa.Core.Node('root')
+        mo = root.addObject("MechanicalObject", name="mo", position=[[1,2,3], [4,5,6]])
+        mo.position.resize(3);
```

Comment:
I might be wrong here, but I think MechanicalObject is a little bit different as it has multiple vectors (x, dx, f, df, etc). Hence it has its own `mo.resize()` (no bindings yet unfortunately). Resizing only the x vector might causes problem later on during the simulation.

Reply:
Hum good point!
This was just an example though, showing how it is possible to resize a DataContainer.
It would be very interesting to add a dedicated binding for MOs that would have its own resize method, and resize position, free_position, reset_position, etc to the same size, and set the `size` datafield to that resize number.

Reply:
Meanwhile, that resize function allows you to implement your own resizing function on the python side for a MechanicalObject

Reply:
> This was just an example though, showing how it is possible to resize a DataContainer.

Of course, I was just thinking out loud when I saw it, nothing to do with your PR :)

> Meanwhile, that resize function allows you to implement your own resizing function on the python side for a MechanicalObject

Indeed, this can be very useful